src/dev/lupio/LupioBLK.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2021 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from m5.objects.Device import BasicPioDevice
+from m5.objects.Device import DmaDevice
+
+from m5.params import Param
+from m5.proxy import Parent
+from m5.defines import buildEnv
+from m5.util.fdthelper import FdtPropertyWords
+
+class LupioBLK(DmaDevice):
+
+    type = 'LupioBLK'
+    cxx_class='gem5::LupioBLK'
+    cxx_header = 'dev/lupio/lupio_blk.hh'
+    pio_size = Param.Addr(0x1000, "PIO Size")
+    image = Param.DiskImage("Disk image")
+    pio_addr = Param.Addr("Device Address")
+    latency = Param.Latency('0ns', "DMA Device Latency")
+    platform = Param.Platform(Parent.any,
+                              "Platform this device is part of.")
+
+    def generateDeviceTree(self, state):
+
+        if buildEnv['TARGET_ISA'] != 'riscv':
+            return
+        node = self.generateBasicPioDeviceNode(state, "lupio-blk",
+                                               self.pio_addr,
+                                               self.pio_size)
+        platform = self.platform.unproxy(self)
+        pic = platform.pic
+        node.append(
+                FdtPropertyWords("interrupts", [platform.lupio_blk_int_id]))
+        node.append(
+                FdtPropertyWords("interrupt-parent", state.phandle(pic)))
+        node.appendCompatible(["lupio,blk"])
+        yield node
```

Comment:
It's nice to have a newline at the end of the file. I suggest configuring your editor to always have exactly one newline at the end of the file and to remove all spaces at the end of lines (it looks like you've done the latter).